detox/src/devices/common/drivers/android/tools/AppInstallHelper.js

```diff
@@ -18,6 +18,28 @@ class AppInstallHelper {
     const binaryPathOnTarget = await this._fileXfer.send(deviceId, binaryPath, binaryFilenameOnTarget);
     await this._adb.remoteInstall(deviceId, binaryPathOnTarget);
   }
+
+  async clearUserData(deviceId, bundleId) {
+    return await this._adb.clearUserData(deviceId, bundleId);
+  }
+
+  async getLocalBinaryHash(binary) {
+    const { getMd5 } = require('./CryptoUtils');
+    await getMd5(binary);
+  }
+
+  async isAlreadyInstalled(deviceId, filehash) {
+    return await this._fileXfer.checkFileExists(deviceId,`${filehash}.hash`);
```

Comment:
This doesn't mean the app is installed. It just means that sometime in the past, something has been installed with `forceAdbInstall` turned off and that the APK was left there...